### PR TITLE
normalizedUri: improved state handling

### PR DESCRIPTION
**Pull request by raymondng**
for: SCRAPED/SCRAPE_FAILED/SCRAPE_WANTED

cc: @ymatsuda

### CODE REVIEW

**Comment by yingjieMiao** [comment]
wow, lots of cases. 

minor comment on style, maybe now it's clearer to write:

uri.state match {
case inactive || active || redirected || unscrapable =>
case scrape_failed || scraped =>
case scrape_wanted
}

**Comment by raymondng** [comment]
Yeah lots of cases :) Thanks for the input!

**Comment by yingjieMiao** [comment]
we can even match on the pair (uri_state, scrape_info), to flatten the nested cases ;-). But I don't worry too much about it.

File: kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala [hunk 1]
```diff
@@ -116,14 +116,14 @@ extends DbRepo[NormalizedURI] with NormalizedURIRepo with ExternalIdColumnDbFunc
...
       }
-    } else if (uri.state == NormalizedURIStates.SCRAPE_FAILED) {
+    } else if (uri.state == NormalizedURIStates.SCRAPE_FAILED || uri.state == NormalizedURIStates.SCRAPED) {
       scrapeRepo.getByUriId(saved.id.get) match { // do not use saveStateAndNextScrape
-        case Some(scrapeInfo) => if (scrapeInfo.state == ScrapeInfoStates.INACTIVE) { scrapeRepo.save(scrapeInfo.withState(ScrapeInfoStates.ACTIVE)) }
-        case Some(scrapeInfo) => // do nothing
-        case None => scrapeRepo.save(ScrapeInfo(uriId = saved.id.get))
+        case Some(scrapeInfo) if (scrapeInfo.state == ScrapeInfoStates.INACTIVE) =>
+          scrapeRepo.save(scrapeInfo.withState(ScrapeInfoStates.ACTIVE))
+        case _ => // do nothing
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/model/NormalizedURIRepo.scala:123]: I see why the integrity test failed. By the old logic, case None => create scrapeInfo.
Here, we assume, if uri si scraped, we always have scrapeInfo for it? 

Line 67-68 of the uriIntegrityTest is just testing some initial status. Clearly it failed by the new logic.

**Pull request merged**